kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -1797,6 +1800,9 @@ private void ungetServiceReferences(final ServiceReference<?>[] refs) {
     protected String marshal(Object object) {
         String result = null;
         ServiceReference<Marshaller>[] marshallerSRs = getXmlMarshallers();
+        if (marshallerSRs == null) {
+            logger.warn("Failed to marshal configuration. No XML marshallers found");
```

Comment:
I believe we could make the same null scenarios in unmarshal(String, Class<T>) and marshal(Object) more consistent. In unmarshal(..), it throws KuraException whereas in marshal(..), it is logged with warning message.